### PR TITLE
Add Select All to entry widget

### LINKED ISSUE #434: Add more shortcuts to fyne.Shortcut
Currently, fyne supports **cut**, **copy** and **paste** shortcuts. That's pretty fine.
The Shortcut interface needs improvements and there's a need for more shortcuts like
1. Undo
2. Redo
3. Select all

### PR DESCRIPTION
**Pull request by spatocode**
**Description**

Improved entry widget by adding select All shortcut

Fixes #434 

**Checklist**

- [x] Tests included
- [x] Lint and formatter run with no errors
- [x] Tests all pass

(where applicable)

- [ ] Public APIs match existing style
- [ ] Any breaking changes have a deprecation path or have been discussed

### CODE REVIEW

File: widget/entry.go [hunk 1]
```diff
@@ -497,6 +497,23 @@ func (e *Entry) pasteFromClipboard(clipboard fyne.Clipboard) {
 	Renderer(e).(*entryRenderer).moveCursor()
 }
 
+// selectAll selects all text in entry
+func (e *Entry) selectAll() {
+	e.Lock()
+	if e.selectKeyDown == false {
```
Comment (andydotxyz) [inline widget/entry.go]: Can you share what this does?
These lines are not tested and I don't understand the purpose

File: widget/entry.go [hunk 2]
```diff
@@ -507,11 +524,14 @@ func (e *Entry) TappedSecondary(pe *fyne.PointEvent) {
 
 	c := fyne.CurrentApp().Driver().CanvasForObject(e)
 
-	item := fyne.NewMenuItem("Paste", func() {
+	pasteItem := fyne.NewMenuItem("Paste", func() {
 		clipboard := fyne.CurrentApp().Driver().AllWindows()[0].Clipboard()
 		e.pasteFromClipboard(clipboard)
 	})
-	e.popUp = NewPopUpMenu(fyne.NewMenu("", item), c)
+	selectAllItem := fyne.NewMenuItem("Select all", func() {
```
Comment (andydotxyz) [inline widget/entry.go]: instead of creating an anonymous function, just pass a.selectAll as the function parameter

**Comment by andydotxyz** [comment]
Thanks for this, I just added a question and a small hint for neater code, I hope it makes sense

File: widget/entry.go (hunk 1 above)
Comment (spatocode) [reply widget/entry.go]: This was to make sure selectAll isn't triggered when shift is held, but i think this is redundant anyway as such scenario wouldn't trigger selectAll. My bad.

**Comment by spatocode** [comment]
> Thanks for this, I just added a question and a small hint for neater code, I hope it makes sense

Yea. No problem! Am just trying to build a side project editor and i think Fyne is the complete Go GUI tookit for that. However, features like these would be helpful moving forward.

Thanks for your great works! I appreciate.

**Comment by awh6al** [comment]
i tried this feature in real world but it seems  it does not working ?

**Comment by andydotxyz** [comment]
To use it you will need to be on the 1.2.0-beta1 pre-release, so either use that module version or check out the tag in your fyne clone inside GOPATH. We will release it all to public in a week or two

**Pull request merged**